app/controllers/products_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :add_to_cart]
+
+  def add_to_cart
+    session[:order_id] ||= Order.create.id
+
+    # this should perhaps be a method in the order model
+    unless Order.find(session[:order_id]).already_has_product?(@product.id)
+      OrderItem.create(product_id: @product.id, order_id: session[:order_id], quantity_ordered: 1)
+    end
+
+    redirect_to cart_path # this should redirect to product show page
```

Comment:
Fun tip: you can mark comments with a `# TODO` flag (like `#TODO: this should redirect...`) and then generate a report of TODOs by running `rake notes:todo`. Works with `#FIXME` and `#OPTIMIZE` too.